### PR TITLE
First steps in test plan

### PR DESCRIPTION
**Pull request by rchiodo**
First steps in a test plan for CTI to follow

### CODE REVIEW

File: testing/scikitlearn/test.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
+import pandas as pd
```
Comment (rchiodo) [inline testing/scikitlearn/test.py:1]: Stole this from Bill's detector.py

File: testing/test_plan.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# Test plan
+
+## Environment
+
+-   OS: XXX (Windows, macOS, latest Ubuntu LTS)
+-   Python
+    -   Distribution: XXX (CPython, miniconda)
+    -   Version: XXX (3.9, 3.10, 3.11)
+-   VS Code: XXX (Insiders)
+-   Python && Pylance Extension: XXX (Insiders)
```
Comment (debonte) [inline testing/test_plan.md:10]: +Jupyter?

File: testing/test_plan.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+-   [ ] Errors
+    1. Open a new python file
+    1. Type invalid python into it. Like `x2+2-3` on a single line.
+    1. Verify you get a red squiggle
+    1. Type valid python after that. 
+    1. Make sure no squiggles happen
+-   [ ] Import errors
+    1. Create a brand new python environment
+    1. Select this in VS code.
```
Comment (debonte) [inline testing/test_plan.md:26]:
```suggestion
    1. Select the new environment in VS code.

```

File: testing/test_plan.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    1. Open a new python file
+    1. Add `import pandas as pd` at the top
+    1. It should error on the pandas
+    1. Open a terminal
+    1. Activate your new environment
```
Comment (debonte) [inline testing/test_plan.md:31]: I think the environment is automatically activated in the terminal? Might be a recent enhancement?

File: testing/test_plan.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    1. Hover over `sys` and make sure it shows something
+    1. Add a new function, like so
+    ```python
+    def foo():
+        '''
+        Doc string for foo 
+        '''
+        pass
+    ```
+    6. Hover over `foo` and make sure you get a doc string
```
Comment (debonte) [inline testing/test_plan.md:53]:
```suggestion
    6. Hover over `foo` and make sure the hover info includes the doc string

```

File: testing/test_plan.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+#### Notebooks
+
+-   [ ] Imports from files
+    1. Open a new jupyter notebook
+    1. Open a new python file
+    1. Create a method in the python file
+    1. In the first cell of the notebook, import the python file
+    1. Verify no red squiggles show up
+    1. Try calling the method from the python file. Something like so:
```
Comment (debonte) [inline testing/test_plan.md:68]: Rephrasing to ensure that "call the method from the python file" is not misinterpreted as "call the method by adding a call in the python file".

```suggestion
    1. In the same cell of the notebook, try calling the method you created. Something like so:

```

File: testing/test_plan.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    7. Goto definition on the method. It should jump to the python file.
+
+
+### P1 Test Scenarios
+
+-   [ ] Rename
+    1. Open a new python file
+    1. Create a method
+    1. Call that method
+    1. Hit F2 on the method and rename it something else
```
Comment (debonte) [inline testing/test_plan.md:83]:
```suggestion
    1. Hit F2 on the method call and rename it something else

```

File: testing/test_plan.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    1. Verify the definition of the method changes
+
+-   [ ] Rename across files
+    1. Open a new python file
+    1. Create a method
+    1. Open another python file
+    1. Reference the first file and its method
```
Comment (debonte) [inline testing/test_plan.md:90]:
```suggestion
    1. Import the method from the first file

```

File: testing/test_plan.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    1. Find all references on the `__init__` function
+    1. Verify it finds the same references as searching for the class name
+
+### P2 Test Scenarios
+
+-   [ ] Scikit learn
+    1. Open VS code
+    1. Pick `Python: Create Environment` command
+    1. Pick `venv`
+    1. After the environment is created, open a terminal and activate it.
```
Comment (debonte) [inline testing/test_plan.md:113]: As above, explicit activation isn't needed?

File: testing/test_plan.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
+# Test plan
+
+## Environment
+
+-   OS: XXX (Windows, macOS, latest Ubuntu LTS)
+-   Python
+    -   Distribution: XXX (CPython, miniconda)
+    -   Version: XXX (3.9, 3.10, 3.11)
```
[the remark this answers is not in the document]
Comment (rchiodo) [reply testing/test_plan.md:8]: Maybe? Can it even be installed?

File: testing/test_plan.md (hunk 4 above)
Comment (rchiodo) [reply testing/test_plan.md:31]: Doesn't always seem to work for me.

File: testing/test_plan.md (hunk 9 above)
Comment (rchiodo) [reply testing/test_plan.md:113]: Just making sure. I'm not sure it always activates.

File: testing/test_plan.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (debonte) [reply testing/test_plan.md:8]: Yes. See right-hand column here: https://www.python.org/downloads/windows/

**Pull request merged**